python/tests/api/test_helpers.py

```diff
@@ -74,7 +74,7 @@ def predicate(self, pred):
             return None
 
 
-class TestDataFile(DataFile):
+class MockDataFile(DataFile):
```

Comment:
Are the renames from `Mock` to `Test` worth the trouble? Unless this is following a python convention, it seems like this is just going to cause git conflicts.

Reply:
Up to you really @rdblue. I may be nit picking but I dislike the visual noise in pytest outputs. Master writes 8-10 lines of warnings in yellow and this patch writes one line in green.

The rename suppresses warnings like below. Given the way pytest aggressively searches for test patterns I think its convention to only have test in the name if you want pytest to try and test it. 

``` bash
tests/api/test_helpers.py:77
  /home/ryan/workspace/iceberg/python/tests/api/test_helpers.py:77: PytestCollectionWarning: cannot collect test class 'TestDataFile' because it has a __init__ constructor (from: tests/api/test_helpers.py)
    class TestDataFile(DataFile):
```

Reply:
It would be preferable to not name them `Test…` as this is the standard prefix for classes that actually have tests.

Reply:
Sounds reasonable to me.